### PR TITLE
Use existing special searching logic for anime

### PR DESCRIPTION
**Pull request by smmalis37**
This change just hooks up the anime searching functions to the existing special searching so that we can at least find some anime specials. This isn't going to find everything, and more work will definitely be needed, but it at least finds some specials. Something is better than nothing right?

### CODE REVIEW

File: src/NzbDrone.Core/IndexerSearch/NzbSearchService.cs [hunk 1]
```diff
@@ -196,10 +196,15 @@ private List<DownloadDecision> SearchAnime(Series series, Episode episode)
             else if (episode.AbsoluteEpisodeNumber.HasValue)
             {
                 searchSpec.AbsoluteEpisodeNumber = episode.AbsoluteEpisodeNumber.Value;
+            }
+            // Season 0 indicates specials, not regular episodes
```
Comment (markus101) [inline src/NzbDrone.Core/IndexerSearch/NzbSearchService.cs]: Comment not required.

File: src/NzbDrone.Core/IndexerSearch/NzbSearchService.cs [hunk 2]
```diff
@@ -216,13 +221,21 @@ private List<DownloadDecision> SearchSpecial(Series series, List<Episode> episod
...
+        private List<DownloadDecision> SearchAnimeSeason(Series series, List<Episode> episodes, int seasonNumber)
         {
-            var downloadDecisions = new List<DownloadDecision>();
-
-            foreach (var episode in episodes)
-            {
-                downloadDecisions.AddRange(SearchAnime(series, episode));
+            var downloadDecisions = new List<DownloadDecision>();
+
+            // Season 0 indicates specials, not regular episodes
```
Comment (markus101) [inline src/NzbDrone.Core/IndexerSearch/NzbSearchService.cs]: Not required either.

**Comment by markus101** [comment]
Some tests to cover the new behavior would be good, namely that we return an error when appropriate and do a special search instead of an error.

File: src/NzbDrone.Core/IndexerSearch/NzbSearchService.cs [hunk 3]
```diff
@@ -216,13 +221,21 @@ private List<DownloadDecision> SearchSpecial(Series series, List<Episode> episod
...
+            if (seasonNumber == 0)
+            {
+                downloadDecisions.AddRange(SearchSpecial(series, episodes));
```
Comment (markus101) [inline src/NzbDrone.Core/IndexerSearch/NzbSearchService.cs]: What about specials that have an absolute episode number, this should search those episodes normally and sure special search if they are missing.

**Comment by markus101** [comment]
Would also be good to have some examples where this is now working properly, so we can test it out.

File: src/NzbDrone.Core/IndexerSearch/NzbSearchService.cs (hunk 3 above)
Comment (smmalis37) [inline src/NzbDrone.Core/IndexerSearch/NzbSearchService.cs]: Do any of those exist? All of the anime I have atm have no absolute episode numbers for all of their specials.

Comment (markus101) [inline src/NzbDrone.Core/IndexerSearch/NzbSearchService.cs]: Yeah they do. Offhand I'm not sure which, asked a couple users to comment here though.

Comment (markus101) [inline src/NzbDrone.Core/IndexerSearch/NzbSearchService.cs]: `One Piece` or `Garo: The Animation` are ones to look at.

**Pull request closed without merging**